privacyidea/lib/caconnectors/msca.py

```diff
@@ -0,0 +1,293 @@
+    """
+
+    connector_type = "microsoft"
+
+    def __init__(self, name, config=None):
+        """
+        Required attributes are:
+         * hostname - the hostname of the middleware
+         * port - the port the middleware listens on
+         * http_proxy - if http proxy should be used
+        :param config: A dictionary with all necessary attributes.
+        :return:
+        """
+        self.name = name
+        self.hostname = None
+        self.port = None
+        self.http_proxy = None
+        self.ca = None
+        self._connection = None
+        if config:
+            self.set_config(config)
+            self._connection = self._connect_to_worker()
+            pass
+        # Note: We can create an empty CA connector object and later configure it using self.set_config().
+
+    @property
+    def connection(self):
+        if not self._connection:
+            self._connection = self._connect_to_worker()
+        return self._connection
+
+    def get_config(self, config):
+        """
+        This method helps to format the config values of the CA Connector.
+        :param config: The configuration as it is stored in the database
+        :type config: dict
+        :return:
+        """
+        config[ATTR.HTTP_PROXY] = is_true(config.get(ATTR.HTTP_PROXY))
+        return config
+
+    def _connect_to_worker(self):
+        """
+        creates a connection to the middleware and returns it.
+        """
+        channel = grpc.insecure_channel('{0!s}:{1!s}'.format(self.hostname, self.port),
+                                        options=(('grpc.enable_http_proxy', int(is_true(self.http_proxy))),))
+        try:
+            grpc.channel_ready_future(channel).result(timeout=3)
+        except grpc.FutureTimeoutError:
+            log.warning("Worker seems to be offline. No connection could be established!")
+        else:
+            return CAServiceStub(channel)
+
+    @classmethod
+    def get_caconnector_description(cls):
+        """
+        Return the description of this CA connectors.
+        This contains the name as a key and the possible parameters.
+
+        :return: resolver description dict
+        :rtype:  dict
+        """
+        typ = cls.connector_type
+        config = {ATTR.HOSTNAME: 'string',
+                  ATTR.PORT: 'string',
+                  ATTR.HTTP_PROXY: 'string',
+                  ATTR.CA: 'string'}
+        return {typ: config}
+
+    def _check_attributes(self):
+        for req_key in [ATTR.HOSTNAME, ATTR.PORT]:
+            if req_key not in self.config:
+                raise CAError("required argument '{0!s}' is missing.".format(
+                    req_key))
+
+    def set_config(self, config=None):
+        self.config = config or {}
+
+        self._check_attributes()
+        self.hostname = self.config.get(ATTR.HOSTNAME)
+        self.port = self.config.get(ATTR.PORT)
+        log.error(config)
```

Comment:
please change this to `log.info` or something else... are we logging the configuration anyway?

Reply:
I remove it. This was actually some remant from some manual tests.